### PR TITLE
Fix entrypoint exiting prematurely when grep fails

### PR DESCRIPTION
**Pull request by lawliet89**
- Also uses `exec` where possible to replace the bash process with whatever
that is being executed

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -38,7 +38,8 @@ gosu "${USER}" sed -i "/\(^max_threads = \).*/ s//\1${AIRFLOW_DAG_CONCURRENCY}/"
 gosu "${USER}" airflow initdb # https://groups.google.com/forum/#!topic/airbnb_airflow/4ZGWUzKkBbw
 
 if [ "$1" = 'afp-scheduler' ]; then
-    (airflow list_dags | grep '^fn_') | while read fn_dag; do
+    echo "Starting scheduler"
```
Comment (chrissng) [inline docker/entrypoint.sh]: nitpick..... don't echo this. it's echoed on line 49

**Review by chrissng — CHANGES_REQUESTED** [review]

File: docker/entrypoint.sh (hunk 1 above)
Comment (lawliet89) [reply docker/entrypoint.sh]: Oops ok.

**Comment by chrissng** [comment]
note to self: `set -euo pipefail` is good practice. more info http://redsymbol.net/articles/unofficial-bash-strict-mode/

**Review by chrissng — APPROVED** [review]
Alright

**Pull request merged**